### ISSUE TITLE
"Languages is invalid"

### ISSUE THREAD

**Issue by error256** [codewars/codewars.com#1291]
In some katas the language list is empty:
![no-languages](https://user-images.githubusercontent.com/5375100/35296260-9041b6c4-008c-11e8-8a21-7829086f7e7d.png)
If I open the dropdown, the languages are there but they aren't selectable. If I try to vote, a message "Languages is invalid" appears in the bottom. It looks like it's just for me and other people don't have this problem.
The first time I encountered this issue a long time ago, since then I've found 3 such katas:
https://www.codewars.com/kata/convert-boolean-values-to-strings-yes-or-no
https://www.codewars.com/kata/jennys-secret-message
https://www.codewars.com/kata/subtract-the-sum
I did complete all of them successfully in multiple languages before the issue appeared.

**Comment by plasticbottle** [comment]
#1323

**Comment by italktotherain** [comment]
I'm seeing this issue with the ruby isograms kata as well

**Comment by Chrono79** [comment]
Seeing this in https://www.codewars.com/kata/pascals-triangle/
And I solved it in javascript and php before (I don't remember if I did submit the final solution tho).

**Comment by Voileexperiments** [comment]
#1666 ?

**Comment by error256** [comment]
There's more variety now: depending on the kata, the sequence of moving between pages and something else, it can be "Languages is invalid", "Failed to cast vote, please try again later", something like "access denied" (I don't remember exactly), and a silent failure with HTTP 405 from challenge_vote. But it's still these 3 katas for me.

**Comment by kazk** [comment]
The `project` document (stores your training sessions and solutions for each language for the kata) is messed up, so anything you do to update this document is failing. The unauthorized error is because Codewars often doesn't know the difference between "you can't do it" and "can't be done" :man_facepalming:

Unfortunately, I don't see anything common in these broken projects so far, so it's difficult to automate the fix. For @error256's case, all 3 happened within 24h.

- 2016-06-17 12:59:49.638 UTC
- 2016-06-18 11:59:19.415 UTC
- 2016-06-18 12:07:26.952 UTC

@error256 Can you try https://www.codewars.com/kata/convert-boolean-values-to-strings-yes-or-no again? It should be fixed.

**Comment by error256** [comment]
> Can you try https://www.codewars.com/kata/convert-boolean-values-to-strings-yes-or-no again? It should be fixed.

Yes, it works now.
It looked like there were no new cases of this bug for a long time, but today @akar-0 mentioned something that looked like this.

**Comment by kazk** [comment]
I fixed all 3 for @error256. Closing this because I don't think this is happening anymore. 

@akar-0 Please let me know if you ran into this for recently solved kata.

If you're experiencing this issue from the past, you can comment here with the following format:

```text
<link to codewars profile>
- <link to kata> (use the URL with kata ID and not slug)
```
```text
https://www.codewars.com/users/Unnamed
- https://www.codewars.com/kata/53369039d7ab3ac506000467
```

And I'll take a look when I have the time.

**Comment by akar-0** [comment]
> @akar-0 Please let me know if you ran into this for recently solved kata.

I solved this kata first in Python a very long time ago (several months ago, maybe more than 1 year). I solved it then in Haskell and Ruby, a long time ago too. I think solved it in Julia more recently, maybe very recently, but I cannot remember for sure (I could be confused with other katas involving BigInt in this language). I realized there was the problem 4 days ago, after forking for update  and approving the Go translation that was published 2 years ago by another user, but I cannot guarantee whether the problem is consecutive to this approval or not.

**Comment by kazk** [comment]
Which kata?

**Comment by akar-0** [comment]
(sorry) [The Millionth Fibonacci Kata](https://www.codewars.com/kata/53d40c1e2f13e331fc000c26/)

**Comment by kazk** [comment]
> I solved it then in Haskell and Ruby, a long time ago too.

Yeah, Ruby was first completed at 2021-10-11. Did you do Ruby again last week (2022-06-10)? Looks like that messed up the data. It somehow partially overwrote the data of the original solution and made it invalid.

I can't fully recover the original data, but made it valid so it shouldn't error anymore.

**Comment by akar-0** [comment]
> Yeah, Ruby was first completed at 2021-10-11. Did you do Ruby again last week (2022-06-10)? Looks like that messed up the data. It somehow partially overwrote the data of the original solution and made it invalid.

I don't think I did Ruby again recently; I am sure of one thing: the Ruby solution that appears in my trainer window currently was not written recently (I would not have used the `.is_even` method). As I told before, if there is one language I made recently, this is most probably Julia.

> I can't fully recover the original data, but made it valid so it shouldn't error anymore.

Yes, the kata works fine now, thanks!

**Comment by kazk** [comment]
Yeah, Julia was completed on 2021-10-15. The Ruby's solution has a completion timestamp 2022-06-10 10:28:21 UTC even though Ruby itself was first completed at 2021-10-11 17:01:51 UTC.

Reopening because it looks like your data was messed up very recently on 2022-06-10. No idea what's causing it.

**Comment by akar-0** [comment]
I didn't mention it before, but this is likely related to the Rust translation I published on 2022-06-10.

**Comment by error256** [comment]
I have this issue with https://www.codewars.com/kata/576400f2f716ca816d001614 now. "language.languageVersions is undefined", HTTP 405 and 422, things like that.
